File: 0485-max-consecutive-ones/0485-max-consecutive-ones.ts
function findMaxConsecutiveOnes(nums: number[]): number {
    let ans = 0, res = 0;
    for (let x of nums) {
        if (x === 0) {
            ans = 0;
        } else {
            ans += 1;
        }
        res = Math.max(res, ans);
    }
    return res;
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int ans = 0, res = 0;
        for(int x : nums)
            res = Math.max(res, ans = x == 0 ? 0 : ans + 1);
        return res;
    }
}